xcdat/dataset.py

```diff
@@ -200,9 +203,14 @@ def open_mfdataset(
 
     References
     ----------
-
     .. [2] https://xarray.pydata.org/en/stable/generated/xarray.open_mfdataset.html
     """
+    if not isinstance(paths, list):
+        if _is_paths_to_xml(paths):
+            paths = _parse_xml_for_nc_glob(paths)  # type: ignore
+        elif os.path.isdir(paths):
+            paths = _parse_dir_for_nc_glob(paths)  # type: ignore
```

Comment:
The main logic addition in `open_mfdataset()` for the `paths` argument being an XML or directory.